src/services/PhotoSearch.js

```diff
@@ -68,6 +70,15 @@ export default async function(onlyFavorites = false, options = {}) {
 			</d:eq>`
 		: ''
 
+	const eqOwner = options.rootFolder === ''
+		? `<d:eq>
+				<d:prop>
+					<oc:owner-id/>
+				</d:prop>
+				<d:literal>${getCurrentUser().uid}</d:literal>
+			</d:eq>`
+		: ''
```

Comment:
This is needed to not have the shares as results.
Why did you need to put a condition on it?
It should not cause any issues like it was before :)

Reply:
The owner filter is only allowed on the root. Dav throws this error if you try on subfolders. What do you think then?

Reply:
Then it should be fixed on server.
cc @rullzer and @icewind1991 :)

Reply:
https://github.com/nextcloud/server/pull/17941

Reply:
https://github.com/nextcloud/server/blob/2a054e6c04e0a40421510eb889cbf59f153c5177/lib/private/Files/Node/Folder.php#L246